drivers/regulator/regulator_pmic.c

```diff
@@ -286,14 +291,15 @@ static int disable_regulator(const struct device *dev)
 
 static int pmic_reg_init(const struct device *dev)
 {
-	struct regulator_config *config = (struct regulator_config *)dev->config;
+	const struct regulator_config *config = dev->config;
+	struct regulator_data *data = dev->data;
 
 	/* Cast the voltage array set at compile time to the voltage range
 	 * struct
 	 */
-	config->voltages = (struct voltage_range *)config->voltage_array;
+	data->voltages = (struct voltage_range *)config->voltage_array;
 	/* Do the same cast for current limit ranges */
-	config->current_levels = (struct current_range *)config->current_array;
+	data->current_levels = (struct current_range *)config->current_array;
 	/* Check to verify we have a valid I2C device */
 	if (config->i2c_dev == NULL || !device_is_ready(config->i2c_dev)) {
```

Comment:
nit: not related, but NULL check here can be removed (device_is_ready supports NULL as an input)

Reply:
Thanks for the heads up, I added a commit to this PR to fix that